openmrs-19/src/main/java/org/motechproject/openmrs19/resource/impl/PatientResourceImpl.java

```diff
@@ -132,44 +115,50 @@ public String getPatientIdentifierTypeUuidByName(String identifierTypeName) thro
     }
 
     @Override
-    public void deletePatient(String uuid) throws HttpException {
-        restfulClient.delete(openmrsInstance.toInstancePathWithParams("/patient/{uuid}?purge", uuid));
+    public void deletePatient(Config config, String uuid) {
+        delete(config, "/patient/{uuid}?purge", uuid);
     }
 
     @Override
-    public Patient updatePatientMotechId(String patientUuid, String newMotechId) throws HttpException {
-        Gson gson = new GsonBuilder().create();
-
-        Identifier patientIdentifier = getPatientIdentifier(patientUuid);
+    public Patient updatePatientMotechId(Config config, String patientUuid, String newMotechId) {
+        Identifier patientIdentifier = getPatientIdentifier(config, patientUuid);
         patientIdentifier.setIdentifier(newMotechId);
-        String identifierUuid = patientIdentifier.getUuid();
-
-        // setting uuid, type and location to null so they are not included in request.
-        patientIdentifier.setUuid(null);
-        patientIdentifier.setIdentifierType(null);
-        patientIdentifier.setLocation(null);
-
-        String requestJson = gson.toJson(patientIdentifier);
-        String responseJson = restfulClient.postForJson(openmrsInstance.toInstancePathWithParams("/patient/{patientUuid}/identifier/{identifierUuid}",
-                patientUuid, identifierUuid), requestJson);
 
+        String requestJson = buildGson().toJson(patientIdentifier);
```

Comment:
do you test this request ? I think here Gson should use option 'excludeFieldsWithoutExposeAnnotation'. Moreover I see that previously also identifierType and location were nulled, can you check how exactly this request should look like